independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java

```diff
@@ -47,22 +48,22 @@ public void writeToDisk() throws IOException {
         if (rootProjectPath != null) {
             Files.write(rootProjectPath.resolve(getSettingsGradlePath()), getModel().getRootSettingsContent().getBytes());
             if (hasRootProjectFile(GRADLE_PROPERTIES_PATH)) {
-                try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-                    getModel().getRootPropertiesContent().store(out, "Gradle properties");
-                    Files.write(rootProjectPath.resolve(GRADLE_PROPERTIES_PATH),
-                            out.toByteArray());
+                try (StringWriter sw = new StringWriter()) {
+                    getModel().getRootPropertiesContent().store(sw, "Gradle properties");
+                    Files.writeString(rootProjectPath.resolve(GRADLE_PROPERTIES_PATH),
+                            sw.toString());
```

Comment:
Using `StringWriter` because of https://github.com/codejive/java-properties/issues/23